src/logic/lemmas.lean

```diff
@@ -0,0 +1,47 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import tactic.split_ifs
+
+/-!
+# More basic logic properties
+
+A few more logic lemmas. These are in their own file, rather than `logic.basic`, because it is
+convenient to be able to use the `split_ifs` tactic.
+-/
+
+variables {α : Type*} {p q r : Prop} [decidable p] [decidable q] {a b c : α}
+
+lemma dite_dite_distrib_left {a : p → α} {b : ¬ p → q → α} {c : ¬ p → ¬ q → α} :
+  dite p a (λ hp, dite q (b hp) (c hp)) =
+    dite q (λ hq, dite p a $ λ hp, b hp hq) (λ hq, dite p a $ λ hp, c hp hq) :=
```

Comment:
I'm guessing you avoided
```suggestion
  (if hp : p then a hp else if hq : q then b hp hq else c hp hq) =
  if hq : q then (if hp : p then a hp else b hp hq) else (if hp : p then a hp else c hp hq) :=
```
because it's less delta-reduced? Perhaps worth an implementation note.

Reply:
Yeah, and because it's frankly quite unreadable.